### DISCUSSION TITLE
Cara memasukkan tabel ke dalam Dashboard

### DISCUSSION THREAD

**Discussion by oktriy** [teamalgoritma/community#83] [Q&A] [answered]
Saya coba memasukkan tabel ke dalam Dashboard, tabel ini adalah hasil dari proses xtabs.
Saat saya tes di Rmd, output nya sudah sesuai (wide), "Year (2019, 2020, 2021)" sudah sebagai kolom masing-masing.
![xtabs Rmd](https://user-images.githubusercontent.com/101442493/167246038-976ca8b4-08f9-4d30-9f01-a73008abfc68.jpg)

tapi setelah di-render dan di Run App, bentuk outputnya berubah (menjadi long).  
![Rproj App](https://user-images.githubusercontent.com/101442493/167246088-ad796e4b-322e-46b5-a77f-8e8935923984.jpg)

**Comment by inytss** [comment, marked as answer]
Dear @oktriy,
Pelampiranoutput xtabs pada Rmd akan berupa sebuah cross section tabel yang mana mengizinkan adanya multiple header maupun index sedangkan apabila kita coba render dalam shiny dengan menggunakan tableOutpur() dan renderTable() maka otomatis output xtabs akan diubah menjadi sebuah data frame biasa yang bentuknya akan memanjang (long format).

Berikut tampilan yang saya coba pada data mtcars.
![Screenshot from 2022-05-09 11-30-53](https://user-images.githubusercontent.com/49224928/167341047-3e0a7cf0-579b-44b7-a551-3c6249881079.png)

Apabila Anda menginginkan outputnya dalam bentuk wide format, Anda dapat mencoba mengkombinasikan fungsi `pivot_wider()`.

Berikut sebagai contoh code yang saya coba pada data `mtcars`.

```
table_mtcars <- xtabs(formula = disp~carb+gear, data = mtcars)
       table_mtcars %>% 
           as.data.frame() %>% 
           tidyr::pivot_wider(names_from = gear, values_from = Freq)
```

Hasilnya akan menjadi seperti ini.
![Screenshot from 2022-05-09 11-32-04](https://user-images.githubusercontent.com/49224928/167341140-5b5acdbc-4114-410a-b97b-2152ddee0fe6.png)

Dimana output xtabs pada Rmd seperti berikut

![Screenshot from 2022-05-09 11-32-54](https://user-images.githubusercontent.com/49224928/167341195-0e11f657-b15b-498e-82d8-e7207fc589e2.png)

Demikian jawaban saya, semoga membantu.

**Reply by oktriy** [reply]
ok, thanks Mba Ina.
